src/http/web_socket.cr

```diff
@@ -100,52 +101,86 @@ class HTTP::WebSocket
     @ws.close(message)
   end
 
-  def run
+  record Ping, message : String
+  record Pong, message : String
+  record Text, message : String
+  record Binary, message : Bytes
+  record Close, message : String
+
+  def handle_opcodes(data)
```

Comment:
This should probably return just `true`/`false` depending if it's a `Ping`/`Pong` or other packet.

It should definitely be a private method.